pkg/rancher-desktop/main/update/__tests__/LonghornProvider.spec.ts

```diff
@@ -1,18 +1,44 @@
 import semver from 'semver';
 
-import { queryUpgradeResponder } from '../LonghornProvider';
+import { queryUpgradeResponder, getWslVersion, UpgradeResponderRequestPayload } from '../LonghornProvider';
 
+import { spawnFile } from '@pkg/utils/childProcess';
 import fetch from '@pkg/utils/fetch';
+
+const itWindows = process.platform === 'win32' ? it : it.skip;
+const itUnix = process.platform !== 'win32' ? it : it.skip;
 
 jest.mock('@pkg/utils/fetch', () => {
   return {
     __esModule: true,
     default:    jest.fn(),
   };
-});
+});
+
+jest.mock('@pkg/utils/childProcess', () => {
+  return {
+    __esModule: true,
+    spawnFile:  jest.fn(),
+  };
+});
 
-describe('queryUpgradeResponder', () => {
+describe('queryUpgradeResponder', () => {
+  afterEach(() => {
+    (spawnFile as jest.Mock).mockReset();
+    (fetch as jest.Mock).mockReset();
+  });
+
   it('should return the latest version', async() => {
+    (spawnFile as jest.Mock).mockResolvedValue({
+      stdout: `WSL version: 1.2.5.0
+Kernel version: 5.15.90.1
+WSLg version: 1.0.51
+MSRDC version: 1.2.3770
+Direct3D version: 1.608.2-61064218
+DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
+Windows version: 10.0.19044.2846
+`,
```

Comment:
Should this be defined as a constant on top? It looks like the same response is used 7 times in this file.